libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -521,6 +521,14 @@ void NavEKF2_core::readGpsData()
                 gpsNotAvailable = false;
             }
 
+            // if we are initialising earth field from tables and we
+            // haven't done that yet then do it now that we have a
+            // position
+            if (!doneFieldFromTables && frontend->_mag_ef_type == 1) {
```

Comment:
Any GPS fix is enough? Seems strange to me that it might not be enough to set the EKF origin but is enough to set earth field - admittedly I'm out of my element here, so I can be very wrong 🙂

Reply:
The field changes only a small amount for even a 100km change in distance and the field tables assume sea level operation, so only a 2D fix is necessary.

Reply:
I will do EK3 once we're happy with the EK2 implementation. From discussions with Paul and flight testing this PR needs quite a bit more work